projects/ngx-translate/core/tests/translate.directive.spec.ts

```diff
@@ -164,19 +166,53 @@ describe('TranslateDirective', () => {
   });
 
   it('should update the DOM when the lang changes and the translation key has line breaks and spaces', () => {
-    expect(fixture.componentInstance.withSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(' TEST ');
+    expect(fixture.componentInstance.withSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(' NESTED.TEST.KEY ');
 
     const en = "This is a test - with trailing spaces in translation key";
     const fr = "C'est un test - avec un espace de fuite dans la clé de traduction";
-    const whiteSpaceFromKey = ' ';
-    translate.setTranslation('en', {"TEST": en});
-    translate.setTranslation('fr', {"TEST": fr});
+    translate.setTranslation("en", { "NESTED": { "TEST": { "KEY": en } } });
+    translate.setTranslation("fr", { "NESTED": { "TEST": { "KEY": fr } } });
 
     translate.use('en');
-    expect(fixture.componentInstance.withSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(whiteSpaceFromKey + en + whiteSpaceFromKey);
+    expect(fixture.componentInstance.withSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(` ${en} `);
 
     translate.use('fr');
-    expect(fixture.componentInstance.withSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(whiteSpaceFromKey + fr + whiteSpaceFromKey);
+    expect(fixture.componentInstance.withSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(` ${fr} `);
+  });
+
+  it('should update the DOM when the lang changes and the translation key is interpolated', async () => {
+    expect(fixture.componentInstance.interpolatedWithSpaceAndLineBreakNoKeyNoParams.nativeElement.innerHTML).toEqual(' Placeholder Text ');
+
+    const en1="This is a test - with trailing spaces in interpolated translation key";
+    const en2="Another test";
+    const fr1="C'est un test - pardon, je ne parle pas francais :)";
+    const fr2="Un autre test";
+    const setKey = key => {
+      fixture.componentInstance.myInterpolatedTranslationKey = key;
+      fixture.componentInstance.changeDetectorRef.markForCheck(); // Needed when changeDetection is ChangeDetectionStrategy.OnPush
```

Comment:
I'm not sure why the test harness is configured with `changeDetection: ChangeDetectionStrategy.OnPush`, as this isn't the default for Angular.  It's fine, but it also means that angular won't detect an interpolated variable change on its own, and we have to tell the change detector when to check for changes.